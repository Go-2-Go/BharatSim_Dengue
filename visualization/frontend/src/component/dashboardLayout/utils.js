import React from 'react';
import renderChart from '../charts/renderChart';

export function renderElement(element) {
  return (
    <div key={element.i} data-grid={element} data-testid={element.i}>
      <div style={{ height: '100%', width: '100%', boxSizing: 'border-box', padding: '10px' }}>
        {renderChart(element.chartType, { config: element.config })}
      </div>
    </div>
  );
}

export function getNewWidgetLayout(numberOfWidgetAdded, cols, count) {
  return createLayout({
    id: `widget-${count}`,
    xPosition: (numberOfWidgetAdded * 2) % cols,
    yPosition: Infinity, // puts it at the bottom
    width: 2,
    height: 2,
  });
}

function createLayout({ id, xPosition, yPosition, width, height }) {
  return {
    i: id,
    x: xPosition,
    y: yPosition,
    w: width,
    h: height,
  };
}
