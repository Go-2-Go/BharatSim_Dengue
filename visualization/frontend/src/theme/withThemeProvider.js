import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme() {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
