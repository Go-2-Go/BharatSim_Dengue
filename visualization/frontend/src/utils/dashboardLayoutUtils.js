/* eslint-disable react/prop-types */

import React from 'react';
import renderChart from '../component/charts/renderChart';

export function renderElement({ layout, chartType, config }) {
  return (
    <div key={layout.i} data-grid={layout} data-testid={layout.i}>
      <div style={{ height: '100%', width: '100%', boxSizing: 'border-box', padding: '10px' }}>
        {renderChart(chartType, { config })}
      </div>
    </div>
  );
}

export function getNewWidgetLayout(numberOfWidgetAdded, cols, count) {
  return createLayout({
    id: `widget-${count}`,
    xPosition: (numberOfWidgetAdded * 2) % cols,
    yPosition: Infinity, // puts it at the bottom
    width: 2,
    height: 2,
  });
}

function createLayout({ id, xPosition, yPosition, width, height }) {
  return {
    i: id,
    x: xPosition,
    y: yPosition,
    w: width,
    h: height,
  };
}
