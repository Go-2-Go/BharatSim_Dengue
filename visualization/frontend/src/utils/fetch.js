import axios from 'axios';
import { httpMethods } from '../constants/fetch';

let shouldShowError;

function commonErrorHandler(err) {
  shouldShowError(true);
}

function initApiConfig({ setError }) {
  shouldShowError = setError;
}

function fetchData({ url, method = httpMethods.GET, headers, data, query, customErrorHandler }) {
  return axios({ url, method, headers, data, params: query })
    .then((res) => res.data)
    .catch((err) => {
      if (customErrorHandler) {
        customErrorHandler(err);
        return;
      }
      commonErrorHandler(err);
    });
}

function uploadData({ url, headers, data, query, customErrorHandler }) {
  return fetchData({
    url,
    method: httpMethods.POST,
    headers,
    data,
    params: query,
    customErrorHandler,
  });
}

export { fetchData, uploadData, initApiConfig };
