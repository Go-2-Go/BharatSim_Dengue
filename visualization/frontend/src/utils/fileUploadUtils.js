const fileUploadedStatus = {
  LOADING: 'loading',
  SUCCESS: 'success',
  ERROR: 'error',
};

function getStatusAndMessageFor(fileUploadStatus, fileName) {
  return {
    [fileUploadedStatus.ERROR]: {
      status: fileUploadedStatus.ERROR,
      message: `Error occurred while unloading ${fileName}`,
    },
    [fileUploadedStatus.SUCCESS]: {
      status: fileUploadedStatus.SUCCESS,
      message: `${fileName} successfully uploaded`,
    },
    [fileUploadedStatus.LOADING]: {
      status: fileUploadedStatus.LOADING,
      message: `uploading ${fileName}`,
    },
  }[fileUploadStatus];
}

export { fileUploadedStatus, getStatusAndMessageFor };
