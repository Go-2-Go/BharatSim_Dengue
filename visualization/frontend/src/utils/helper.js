function updateState(prevState,updatedValue){
  return Object.assign({},prevState,updatedValue)
}

export {
  updateState,
}