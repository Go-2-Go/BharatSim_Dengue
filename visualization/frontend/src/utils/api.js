import { contentTypes, httpMethods } from '../constants/fetch';
import { url as serviceURL } from './url';
import { fetchData, uploadData } from './fetch';

function headerBuilder({ contentType }) {
  return {
    'content-type': contentType,
  };
}

function formDataBuilder(data) {
  const formData = new FormData();
  data.forEach(({ name, value }) => formData.append(name, value));
  return formData;
}

const api = {
  saveDashboard: async ({ widgets, layout, dashboardId, name, count }) => {
    return uploadData({
      url: serviceURL.DASHBOARD_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      data: JSON.stringify({
        dashboardData: { widgets, layout, dashboardId, name, count },
      }),
    });
  },

  getAllDashBoard: async () => fetchData({ url: serviceURL.DASHBOARD_URL }),

  uploadFileAndSchema: async ({ file, schema }) =>
    uploadData({
      url: serviceURL.DATA_SOURCES,
      data: formDataBuilder([
        { name: 'datafile', value: file },
        { name: 'schema', value: JSON.stringify(schema) },
      ]),
      headers: headerBuilder({ contentType: contentTypes.FILE }),
    }),

  getCsvHeaders: async (dataSourceId) => fetchData({ url: serviceURL.getHeaderUrl(dataSourceId) }),

  getDatasources: async () => fetchData({ url: serviceURL.DATA_SOURCES }),

  getData: async (datasource, query) =>
    fetchData({ url: serviceURL.getDataUrl(datasource), query }),

  saveProject: async ({ id, ...data }) => {
    const requestObject = {
      url: serviceURL.PROJECT_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
    };
    if (id) {
      return uploadData({
        ...requestObject,
        data: JSON.stringify({ projectData: { ...data, id } }),
        method: httpMethods.PUT,
      });
    }
    return uploadData({
      ...requestObject,
      data: JSON.stringify({ projectData: { ...data } }),
      method: httpMethods.POST,
    });
  },

  getProjects: async () => {
    return fetchData({
      url: serviceURL.PROJECT_URL,
    });
  },
  getProject: async (id) => {
    return fetchData({
      url: serviceURL.getProjectUrl(id),
    });
  },
};

export { api };
