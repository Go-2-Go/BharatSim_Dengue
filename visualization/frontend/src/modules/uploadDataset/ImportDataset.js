import React from 'react';
import PropTypes from 'prop-types';

import Box from '@material-ui/core/Box';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import fileImportIcon from '../../assets/images/importFileIcon.svg';
import { parseCsv } from '../../utils/fileUploadUtils';
import LoaderOrError from '../../component/loaderOrError/LoaderOrError';
import useLoader, { loaderStates } from '../../hook/useLoader';

const useStyles = makeStyles((theme) => {
  return {
    importDataContainer: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center',
    },
    textContainer: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      marginTop: theme.spacing(5),
    },
  };
});

function ImportDataset({ setFile, handleNext, setPreviewData, setErrorStep }) {
  const classes = useStyles();
  const { loadingState, stopLoaderAfterSuccess, stopLoaderAfterError, startLoader } = useLoader(
    loaderStates.SUCCESS,
  );

  function onParse(csvData) {
    const { data, errors } = csvData;
    setPreviewData(data);
    if (errors.length > 0) {
      stopLoaderAfterError('unable to parse dataset');
      setErrorStep(0);
      return;
    }
    stopLoaderAfterSuccess();
    handleNext();
  }

  function handleFileImport(event) {
    startLoader();
    const selectedFile = event.target.files[0];
    parseCsv(selectedFile, onParse);
    setFile(selectedFile);
  }

  return (
    <LoaderOrError loadingState={loadingState.state}>
      <Box mb={8} mx={14} py={8} className={classes.importDataContainer}>
        <img src={fileImportIcon} alt="Import File" />
        <Box className={classes.textContainer}>
          <Typography>Drag your file here or </Typography>
          <Box ml={3}>
            <Button variant="outlined" component="label">
              Browse
              <input
                data-testid="file-input"
                type="file"
                style={{ display: 'none' }}
                onChange={handleFileImport}
              />
            </Button>
          </Box>
        </Box>
      </Box>
    </LoaderOrError>
  );
}

ImportDataset.propTypes = {
  setFile: PropTypes.func.isRequired,
  handleNext: PropTypes.func.isRequired,
  setPreviewData: PropTypes.func.isRequired,
  setErrorStep: PropTypes.func.isRequired,
};

export default ImportDataset;
