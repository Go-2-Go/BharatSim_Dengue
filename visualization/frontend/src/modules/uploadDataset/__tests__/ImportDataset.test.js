import React from 'react';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import withThemeProvider from '../../../theme/withThemeProvider';
import ImportDataset from '../ImportDataset';
import * as fileUtils from '../../../utils/fileUploadUtils';

jest.spyOn(fileUtils, 'parseCsv').mockImplementation((csvFile, onComplete) => {
  const data = { data: [{ col1: 'row1', col2: 1 }], errors: [] };
  onComplete(data);
});

describe('Upload Dataset', () => {
  const Component = withThemeProvider(ImportDataset);
  let setFileMock;
  let handleNextMock;
  let setPreviewDataMock;
  let setErrorStepMock;
  beforeEach(() => {
    setFileMock = jest.fn();
    handleNextMock = jest.fn();
    setPreviewDataMock = jest.fn();
    setErrorStepMock = jest.fn();
  });

  it('should match snapshot for import dataset component', () => {
    const { container } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should import and set csv file ', () => {
    const { getByTestId } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, { target: { files: [{ name: 'csv', size: '10' }] } });

    expect(setFileMock).toHaveBeenCalledWith({ name: 'csv', size: '10' });
  });
  it('should import and parse csv file ', () => {
    const { getByTestId } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, { target: { files: [{ name: 'csv', size: '10' }] } });

    expect(fileUtils.parseCsv).toHaveBeenCalledWith(
      { name: 'csv', size: '10' },
      expect.any(Function, () => {}),
    );
  });
  it('should set preview data for given file', () => {
    const { getByTestId } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, { target: { files: [{ name: 'csv', size: '10' }] } });

    expect(setPreviewDataMock).toHaveBeenCalledWith([{ col1: 'row1', col2: 1 }]);
  });
  it('should set error for any parsing error in given file', () => {
    jest.spyOn(fileUtils, 'parseCsv').mockImplementation((csvFile, onComplete) => {
      const data = { data: [{ col1: 'row1', col2: 1 }], errors: ['error'] };
      onComplete(data);
    });
    const { getByTestId } = render(
      <Component
        setFile={setFileMock}
        handleNext={handleNextMock}
        setPreviewData={setPreviewDataMock}
        setErrorStep={setErrorStepMock}
      />,
    );
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, { target: { files: [{ name: 'csv', size: '10' }] } });

    expect(setErrorStepMock).toHaveBeenCalledWith(0);
  });
});
