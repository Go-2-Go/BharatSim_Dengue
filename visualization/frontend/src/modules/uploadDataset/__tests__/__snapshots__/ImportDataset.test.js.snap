// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload Dataset should match snapshot for import dataset component 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-importDataContainer-1"
  >
    <img
      alt="Import File"
      src="importFileIcon.svg"
    />
    <div
      class="MuiBox-root MuiBox-root-4 makeStyles-textContainer-2"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Drag your file here or 
      </p>
      <div
        class="MuiBox-root MuiBox-root-5"
      >
        <label
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Browse
            <input
              data-testid="file-input"
              style="display: none;"
              type="file"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </label>
      </div>
    </div>
  </div>
</div>
`;
